### PR TITLE
introducing WriteService in Sandbox

### PR DESCRIPTION
**Pull request by gaboraranyossy-da**
See title.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  /**
+    * Validates that the given ledger effective time is within an acceptable time window of the current system time.
+    *
+    * @param currentTime              the current time
+    * @param givenLedgerEffectiveTime The ledger effective time to validate.
+    * @param givenMaximumRecordTime   The maximum record time to validate.
+    * @return true if successful
+    */
+  def checkLet(
```
Comment (dajmaki) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala]: what do you think about just having the "TimeModel" data here and have the checks separate from it? e.g. https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto#L127.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala [hunk 2]
```diff
@@ -89,9 +91,10 @@ trait WriteService {
     *   transaction id.
     *
     */
+  //TODO: update docs
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
-      transaction: SubmittedTransaction): Unit
+      transaction: SubmittedTransaction): Future[SubmissionResult]
```
Comment (dajmaki) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]: as discussed, lets use java-friendly types at this level, e.g. CompletableFuture.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala (hunk 1 above)
Comment (gaboraranyossy-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala]: good idea, will do that.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala (hunk 2 above)
Comment (gaboraranyossy-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]: changed it to use the interface `CompletionStage`

**Pull request merged**